com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -706,29 +736,59 @@ private void ApplyAuthoritativeState()
                     interpolatedRotAngles.z = isTeleporting || !Interpolate ? networkState.RotAngleZ : eulerAngles.z;
                 }
             }
+            else if (Interpolate && SynchronizeRotation)
+            {
+                var eulerAngles = m_RotationInterpolator.GetInterpolatedValue().eulerAngles;
+                interpolatedRotAngles = eulerAngles;
+                if (SyncRotAngleY)
+                {
+                    interpolatedRotAngles.x = m_RotationInterpolator.GetInterpolatedValue().eulerAngles.x;
+                }
+
+                if (SyncRotAngleY)
+                {
+                    interpolatedRotAngles.y = m_RotationInterpolator.GetInterpolatedValue().eulerAngles.y;
+                }
+
+                if (SyncRotAngleY)
+                {
+                    interpolatedRotAngles.y = m_RotationInterpolator.GetInterpolatedValue().eulerAngles.y;
+                }
+            }
 
             // Update all scale axis that were changed in this state update
             if (networkState.HasScaleX)
             {
                 interpolatedScale.x = isTeleporting || !Interpolate ? networkState.ScaleX : m_ScaleXInterpolator.GetInterpolatedValue();
             }
+            else if (Interpolate && SyncScaleX)
+            {
+                interpolatedScale.x = m_ScaleXInterpolator.GetInterpolatedValue();
+            }
 
             if (networkState.HasScaleY)
             {
                 interpolatedScale.y = isTeleporting || !Interpolate ? networkState.ScaleY : m_ScaleYInterpolator.GetInterpolatedValue();
             }
+            else if (Interpolate && SyncScaleY)
+            {
+                interpolatedScale.y = m_ScaleYInterpolator.GetInterpolatedValue();
+            }
 
             if (networkState.HasScaleZ)
             {
                 interpolatedScale.z = isTeleporting || !Interpolate ? networkState.ScaleZ : m_ScaleZInterpolator.GetInterpolatedValue();
             }
+            else if (Interpolate && SyncScaleZ)
+            {
+                interpolatedScale.z = m_ScaleZInterpolator.GetInterpolatedValue();
+            }
 
             // Apply the new position
-            if (networkState.HasPositionChange)
+            if (networkState.HasPositionChange || Interpolate && SynchronizePosition)
```

Comment:
Little bit of a picky comment... I know it's well-defined in the C# standard, but as someone reading the code, operator precedence between `||` and `&&` isn't something I intuitively know... is this `(networkState.HasPositionChange || Interpolate) && SynchronizePosition` or `networkState.HasPositionChange || (Interpolate && SynchronizePosition)`? Can you add parentheses to all of these spots where both `||` and `&&` are used to make sure it's clearly communicated (to other developers) which order you're expecting them to be evaluated in?

(This is actually something I'd love if our style could enforced... I've seen a lot of devs - myself included - get bitten by the compiler's order of operations being different than what they were thinking when they wrote the code.)

Reply:
I can add that but it will get flagged as "un-needed" parenthesis.
But the general C# Boolean rule is:   && is always handled before ||
(sort of like x + y * 5... where * always happens before +)

Reply:
Huh... that's kind of frustrating, to be honest, that it would flag parentheses added for readability as an issue.

Even knowing the order of operations, I do tend to always write `x + (y * 5)` because the parentheses don't hurt anything and they remove the need to think about the order when reading it - the order's obvious.

Reply:
Yeah...I sort of like having parenthesis like that as well...you typically see that more in C++ than C#.
Now, when a parenthesis is needed and won't get flagged:
- `(x + y) * 5`
- `(Something || SomethingElse) && MustBeTrueThing)`

Reply:
This preference may indeed be a result of C++ being my native language. :P Most C++ compilers actually flag a warning (at high warning levels) if you DON'T use parentheses in a statement combining `||` and `&&`

Reply:
I could be wrong about this...

Reply:
I am wrong about this.
It is "ok" because it is an order of operations thing (need to dust off that part of my brain).

> But the general C# Boolean rule is: && is always handled before ||
This is absolutely wrong (my bad).

The thing I was thinking about was how it evaluates the expression (left to right)...Jesse helped clear me up on this.  :)
However, the good news is that one particular section (I evidently was completely not thinking when I typed my response) can have parenthesis without any issues... so I will add those where it is applicable.